server2/Arcadia.Assistant/Arcadia.Assistant.Notifications/Notifications.cs

```diff
@@ -0,0 +1,154 @@
+
+    using Interfaces;
+
+    using Microsoft.Extensions.Logging;
+    using Microsoft.ServiceFabric.Services.Communication.Runtime;
+    using Microsoft.ServiceFabric.Services.Runtime;
+
+    using Models;
+
+    using PushNotifications.Contracts;
+    using PushNotifications.Contracts.Models;
+
+    /// <summary>
+    ///     An instance of this class is created for each service instance by the Service Fabric runtime.
+    /// </summary>
+    public sealed class Notifications : StatelessService, INotifications
+    {
+        private readonly IDeviceRegistry deviceRegistry;
+        private readonly ILogger logger;
+
+        private readonly Dictionary<CalendarEventType, IEnumerable<NotificationProviderType>> notificationProvidersMap;
+        private readonly INotificationSettings notificationSettings;
+        private readonly IPushNotifications pushNotifications;
+
+        public Notifications(
+            StatelessServiceContext context,
+            INotificationSettings notificationSettings,
+            IDeviceRegistry deviceRegistry,
+            IPushNotifications pushNotifications,
+            ILogger logger)
+            : base(context)
+        {
+            this.notificationSettings = notificationSettings;
+            this.deviceRegistry = deviceRegistry;
+            this.pushNotifications = pushNotifications;
+            this.logger = logger;
+            this.notificationProvidersMap = new Dictionary<CalendarEventType, IEnumerable<NotificationProviderType>>
+            {
+                {
+                    CalendarEventType.Vacation, new List<NotificationProviderType>
+                        { NotificationProviderType.Push }
+                },
+                {
+                    CalendarEventType.Dayoff, new List<NotificationProviderType>
+                        { NotificationProviderType.Push }
+                },
+                {
+                    CalendarEventType.Workout, new List<NotificationProviderType>
+                        { NotificationProviderType.Push }
+                },
+                {
+                    CalendarEventType.Sickleave, new List<NotificationProviderType>
+                        { NotificationProviderType.Push }
+                }
+            };
+        }
+
+        public async Task Send(IEnumerable<string> employeeIds, NotificationType notificationType, CalendarEventType eventType, NotificationMessage notificationMessage, CancellationToken cancellationToken)
+        {
+            if (employeeIds.IsNullOrEmpty() || !this.notificationProvidersMap.ContainsKey(eventType))
+            {
+                return;
+            }
+
+            var notificationProviders = this.notificationProvidersMap[eventType];
+            foreach (var providerType in notificationProviders)
+            {
+                switch (providerType)
+                {
+                    case NotificationProviderType.Push:
+                        if (this.notificationSettings.EnablePush)
```

Comment:
why switchCase if you have input arguments defined. Just have an interface and implementations?

Reply:
we have several providers type... Push and Email - we discussed it.